src/test/java/org/broadinstitute/consent/http/service/dao/DarCollectionServiceDAOTest.java

```diff
@@ -95,26 +109,136 @@ public void testCreateElectionsForDarCollectionManualReview() throws Exception {
+   *  - Collection has 2 DAR/Dataset combinations
+   *  - User is a DAC chair for only one of the DAR/Dataset combinations
+   *  - Elections created should only be for the DAR/Dataset for the user
+   */
+  @Test
+  public void testCreateElectionsForDarCollectionWithMultipleDatasetsForChair() throws Exception {
+    initService();
+
+    // Start off with a collection and a single DAR
+    DarCollection collection = setUpDarCollectionWithDacDataset();
+    Optional<DataAccessRequest> dar = collection.getDars().values().stream().findFirst();
+    assertTrue(dar.isPresent());
+    assertFalse(dar.get().getData().getDatasetIds().isEmpty());
+    Integer datasetId = dar.get().getData().getDatasetIds().get(0);
+    assertNotNull(datasetId);
+    Optional<Dac> dac = dacDAO.findDacsForDatasetIds(List.of(datasetId)).stream().findFirst();
+    assertTrue(dac.isPresent());
+    List<User> dacUsers = dacDAO.findMembersByDacId(dac.get().getDacId());
+    Optional<User> chair = dacUsers.stream().filter(u -> u.hasUserRole(UserRoles.CHAIRPERSON)).findFirst();
+    assertTrue(chair.isPresent());
+
+    // Add another DAR with Dataset to collection.
+    // This one should not be available to the chairperson created above.
+    DataAccessRequest dar2 = addDARWithDacAndDatasetToCollection(collection);
+    // refresh the collection
+    collection = darCollectionDAO.findDARCollectionByCollectionId(collection.getDarCollectionId());
+
+    serviceDAO.createElectionsForDarCollection(chair.get(), collection);
+
+    List<Election> createdElections =
+        electionDAO.findLastElectionsByReferenceIds(List.of(dar.get().getReferenceId()));
+    List<Vote> createdVotes =
+        voteDAO.findVotesByElectionIds(
+            createdElections.stream().map(Election::getElectionId).collect(Collectors.toList()));
+
+    // Ensure that we have an access and rp election
+    assertFalse(createdElections.isEmpty());
+    assertTrue(createdElections.stream().anyMatch(e -> e.getElectionType().equals(ElectionType.DATA_ACCESS.getValue())));
+    assertTrue(createdElections.stream().anyMatch(e -> e.getElectionType().equals(ElectionType.RP.getValue())));
+    // Ensure that we have primary vote types
+    assertFalse(createdVotes.isEmpty());
+    assertTrue(createdVotes.stream().anyMatch(v -> v.getType().equals(VoteType.CHAIRPERSON.getValue())));
+    assertTrue(createdVotes.stream().anyMatch(v -> v.getType().equals(VoteType.FINAL.getValue())));
+    assertTrue(createdVotes.stream().anyMatch(v -> v.getType().equals(VoteType.DAC.getValue())));
+    assertTrue(createdVotes.stream().anyMatch(v -> v.getType().equals(VoteType.AGREEMENT.getValue())));
+
+    // Make sure we did not create elections for the DAR/Dataset that the chair does not have access to.
+    List<Election> nonCreatedElections =
+        electionDAO.findLastElectionsByReferenceIds(List.of(dar2.getReferenceId()));
+    assertTrue(nonCreatedElections.isEmpty());
+  }
+
+  /**
+   * This test covers the case where:
```

Comment:
Can you add a chair version of this test? (A test that ensures that a Chair can re-open elections)

Reply:
Good call 👍🏽

Reply:
@JVThomas - added a new test for this case. PTAL thank you!